src/main/java/backend/IssueMetadata.java

```diff
@@ -33,6 +33,15 @@ public IssueMetadata(IssueMetadata other) {
         this.isUpdated = other.isUpdated;
     }
 
+    // Copy constructor used in reconciliation
+    public IssueMetadata(IssueMetadata other, boolean isUpdated) {
+        this.events = new ArrayList<>(other.events);
+        this.comments = new ArrayList<>(other.comments);
+        this.nonSelfUpdatedAt = other.nonSelfUpdatedAt;
+        this.nonSelfCommentCount  = other.nonSelfCommentCount;
+        this.isUpdated = isUpdated;
```

Comment:
We should document what `isUpdated` does (controls whether `nonSelfUpdatedAt` or `updatedAt` is used).